src/Components/Aspire.Nats.Client/NatsHealthCheck.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using NATS.Client.Core;
+
+namespace Aspire.Nats.Client;
+
+public class NatsHealthCheck(INatsConnection connection) : IHealthCheck
+{
+    public Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+    {
+        return Task.FromResult(connection.ConnectionState == NatsConnectionState.Open
+            ? HealthCheckResult.Healthy()
+            : HealthCheckResult.Unhealthy());
+    }
```

Comment:
https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/73abc7a094f9ef625b83455898a21d22d2fa6a82/src/HealthChecks.Nats/NatsHealthCheck.cs#L74-L83 has more complicated logic than this. Is this enough to be able to tell if the connection is healthy?

Reply:
updated to be a bit more granular i.e. added degraded state. But description and stats only exist on the old client Nats.Client v1. One we're using here is v2 and it's a rewrite and most of that is not exposed yet.